nodepool/config/templates/stack-template.json

```diff
@@ -8,8 +8,50 @@
         "TargetCapacity": {{$.Worker.SpotFleet.TargetCapacity}},
         "SpotPrice": "{{$.Worker.SpotFleet.SpotPrice}}",
         "LaunchSpecifications": [
+          {{if .Worker.TopologyPrivate}}
+          {{range $subnetIndex, $subnet := $.Worker.PrivateSubnets}}
```

Comment:
Sorry for being silent for a while but yes, I believe you did a great job implementing my vague idea!
Would you mind going ahead with your work for controller and etcd, too?